fedimint-core/src/db/mod.rs

```diff
@@ -578,6 +602,19 @@ impl<'a> ISingleUseDatabaseTransaction<'a> for CommittableIsolatedDatabaseTransa
         Ok(Box::pin(stream::iter(stream)))
     }
 
+    async fn raw_find_by_prefix_sorted_descending(
+        &mut self,
+        key_prefix: &[u8],
+    ) -> Result<PrefixStream<'_>> {
+        let mut isolated = IsolatedDatabaseTransaction::new(self.dbtx.as_mut(), Some(self.prefix));
+        let stream = isolated
+            .raw_find_by_prefix_sorted_descending(key_prefix)
+            .await?
+            .collect::<Vec<_>>()
```

Comment:
This seems to collect into a `Vec`, which seems undesirable. I see that `raw_find_by_prefix` was doing it already, but isn't this ruining the benefits of streaming? We are still going to load all the matching notes at once into memory, just fake later that it was streaming ...

Reply:
Nice catch, this was an oversight.

Reply:
This is trickier than it seems. The borrow checker isn't happy if you remove the collect. I know it is feasible because if I inline the code of `IsolatedDatabaseTransaction` then it works:  https://github.com/fedimint/fedimint/pull/2188

@m1sterc001guy any ideas here?

Reply:
This is where I got stuck too :) Unless @dpc has ideas I think its fine to just inline the isolated code here

Reply:
Will be addressed on https://github.com/fedimint/fedimint/pull/2202

Reply:
In fact it's blocked by that PR because `raw_find_by_prefix_sorted_descending` will be implemented in the same way

Reply:
This requires a rebase now I assume?